packages/common/src/search/wellKnownCatalog.ts

```diff
@@ -40,6 +43,8 @@ export interface IGetWellKnownCatalogOptions {
   collectionNames?: WellKnownCollection[];
   /** additional filters to apply to the catalog scope */
   filters?: IFilter[];
+  /** optional context */
+  context?: IArcGISContext;
```

Comment:
I'm wondering if this should be pulled out into one of the top-level args for the `getWellKnownCatalog` function. Anything in `options` should really be optional since we're using the same interface for all catalogs...and `context` is technically required for the "community" + "partner" catalogs. As an aside, I also think `context` removes the need to pass in `user` as an option (which also doesn't _really_ appear to be optional). 

We could also try to do something with discriminated unions depending on the `catalogName` 🤔 but in any case, I still think `context` should be a required arg for this function. Unfortunately even if we pull it out into a top-level arg, we'll still have to make it optional to avoid a breaking change. Maybe we can leave a note to change this in the next breaking change though.

Reply:
As discussed I've added two TODO's to the interface and to the actual function.